tao_compiler/mlir/disc/transforms/tests/cpu-only-sparse-segment-mean-lhlo-legalize-roots-to-loops.mlir

```diff
@@ -0,0 +1,28 @@
+// RUN: DISC_ENABLE_SHAPE_CONSTRAINT_IR=0 DISC_ENABLE_HORIZONTAL_FUSION=0 disc-opt %s -disc-lhlo-legalize-roots-to-parallel-loops -split-input-file | FileCheck %s
+
+// CHECK-LABEL: @sparse_segment_mean
+// CHECK-SAME: (%[[INPUT1:.*]]: memref<?x?xf32, "cpu">, %[[INPUT2:.*]]: memref<?xi32, "cpu">, %[[INPUT3:.*]]: memref<?xi32, "cpu">, %[[OUT1:.*]]: memref<?x?xf32, "cpu">) -> memref<?x?xf32, "cpu">
+func.func @sparse_segment_mean(
+    %input1: memref<?x?xf32, "cpu">,
+    %input2: memref<?xi32, "cpu">,
+    %input3: memref<?xi32, "cpu">,
+    %out1: memref<?x?xf32, "cpu">
+    ) -> (
+    memref<?x?xf32, "cpu">
+  ) {
+  // CHECK-NOT lmhlo
```

Comment:
Is the following check strong enough?

Reply:
Current impl has 2 major loops(the one doing memset not included), this checks for the most complex one which dose the naive mean computation.